### PR TITLE
feat(ecs): secret JSON field for Fargate tasks

### LINKED ISSUE #11341: [aws-ecs] Fargate launch type now supports secretmanager parameter names
As mentioned in [this issue](https://github.com/aws/containers-roadmap/issues/385) and [here](https://aws.amazon.com/about-aws/whats-new/2020/11/aws-fargate-for-amazon-ecs-launches-features-focused-on-configuration-and-metrics/) AWS ECS Fargate now supports fetching a specific json field in a secret. 

The CDK ecs module supports specifying a field in `Secret.fromSecretsManager(secret, field?)` method, but if you specify a field when using the Fargate launch type then you currently get an error message that this is not supported `Cannot specify secret JSON field for a task using the FARGATE launch type: 'SECRET' in container 'container'`. 

It would be really nice if this new ECS feature could make it into CDK quickly!

### Use Case

I want to keep one secretsmanager secret with f.e 'my-container-env' and use that secret with various fields to specify my container environment for fargate tasks, instead of having to use ssm or one secret for each value.

### Proposed Solution

Not sure. The resources I found dont say whether this is supported in CloudFormation too right away. If it is, maybe you could just start allowing this 'field' to be present in fargate tasks too!

### Other

* [ ] :wave: I may be able to implement this feature request
* [ ] :warning: This feature might incur a breaking change

---

This is a :rocket: Feature Request

### PR DESCRIPTION
**Pull request by jogold**
Fargate tasks running on platform version 1.4 can now reference a
specific JSON field of a secret stored in Secrets Manager.

Remove the error in `ContainerDefinition` and add a check on the
platform version in `FargateService`.

See https://aws.amazon.com/about-aws/whats-new/2020/11/aws-fargate-for-amazon-ecs-launches-features-focused-on-configuration-and-metrics/

See https://github.com/aws/containers-roadmap/issues/385#issuecomment-722696672

Closes #11341

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/@aws-cdk/aws-ecs/lib/fargate/fargate-service.ts [hunk 1]
```diff
@@ -141,6 +141,12 @@ export class FargateService extends BaseService implements IFargateService {
       throw new Error('Only one of SecurityGroup or SecurityGroups can be populated.');
     }
 
+    if (props.taskDefinition.referencesSecretJsonField
+        && props.platformVersion
+        && SECRET_JSON_FIELD_UNSUPPORTED_PLATFORM_VERSIONS.includes(props.platformVersion)) {
+      throw new Error('The task definition of this service uses at least a container that references a secret JSON field. This feature requires platform version 1.4.0 or later.');
```
Comment (SoManyHs) [inline packages/@aws-cdk/aws-ecs/lib/fargate/fargate-service.ts:147]: Just a nit: 

```suggestion
      throw new Error(`The task definition of this service uses at least one container that references a secret JSON field. This feature requires platform version ${FargatePlatformVersion.VERSION1_4} or later.`);
```

**Review by SoManyHs — COMMENTED** [review]
One small nit, but otherwise LGTM! Thanks for the speedy turnaround!

**Review by SoManyHs — APPROVED** [review]
🎉 Thanks @jogold !!

**Comment by jogold** [comment]
@SoManyHs Can you reapprove? You merged from master after reviewing, this dismisses the review.

**Comment by jogold** [comment]
@SoManyHs ping

**Comment by marcotesch** [comment]
@SoManyHs an approval would be much appreciated, so we can use the feature within our deployments :-)

**Comment by jogold** [comment]
@marcotesch you can use the following workaround in the meantime https://github.com/aws/aws-cdk/issues/11341#issuecomment-723553936

**Comment by SoManyHs** [comment]
> @SoManyHs Can you reapprove? You merged from master after reviewing, this dismisses the review.

Ah, didn't see that. Sorry about the delay!

**Review by SoManyHs — APPROVED** [review]

**Pull request merged**